exercises/t_jubb/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,270 @@
+
```

Comment:
This is the type of problem where we should start splitting code into separate header and implementation files. For instance we would have separate header files for `Shape` and all of the derived classes and also `ShapeSorter`. Usually C++ code is split between a header file  (`.h`) for declarations and an implementation file (`*.cpp`) - this helps maintain some sort of order as the code base grows.